<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SpectrumViewer</class>
 <widget class="QWidget" name="SpectrumViewer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>614</width>
    <height>426</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>400</width>
    <height>300</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Spectrum Viewer</string>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QToolBar" name="MainToolBar">
     <property name="windowTitle">
      <string>toolBar</string>
     </property>
     <property name="movable">
      <bool>false</bool>
     </property>
     <property name="floatable">
      <bool>false</bool>
     </property>
     <addaction name="action_Save_CSV"/>
     <addaction name="separator"/>
     <addaction name="action_Run"/>
     <addaction name="action_Stop"/>
     <addaction name="separator"/>
     <addaction name="action_SilentMode"/>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="mainFrame">
     <property name="leftMargin">
      <number>9</number>
     </property>
     <property name="rightMargin">
      <number>9</number>
     </property>
     <property name="bottomMargin">
      <number>9</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="spacing">
        <number>6</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>6</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>6</number>
       </property>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QLabel" name="LExposureTime">
           <property name="text">
            <string>Exposure time</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="SBExposureTime">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>6</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="LLowerFrequency">
           <property name="text">
            <string>Lower limit</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="SBLowerFrequency">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="decimals">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>6</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="LUpperFrequency">
           <property name="text">
            <string>Upper limit</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="SBUpperFrequency">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="decimals">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QChartView" name="Spectrum">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>0</height>
          </size>
         </property>
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <property name="spacing">
        <number>12</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="LState">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>2</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>22</height>
          </size>
         </property>
         <property name="text">
          <string>TextLabel</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="PBProgress">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>140</width>
           <height>0</height>
          </size>
         </property>
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
  <action name="action_Save_CSV">
   <property name="icon">
    <iconset resource="../../DynExpManager.qrc">
     <normaloff>:/DynExpManager/icons/Document-save.svg</normaloff>:/DynExpManager/icons/Document-save.svg</iconset>
   </property>
   <property name="text">
    <string>Save CSV</string>
   </property>
   <property name="toolTip">
    <string>Save CSV</string>
   </property>
  </action>
  <action name="action_Run">
   <property name="icon">
    <iconset resource="../../DynExpManager.qrc">
     <normaloff>:/DynExpManager/icons/Media-playback-start.svg</normaloff>:/DynExpManager/icons/Media-playback-start.svg</iconset>
   </property>
   <property name="text">
    <string>Run once</string>
   </property>
   <property name="toolTip">
    <string>Record single spectrum</string>
   </property>
   <property name="shortcut">
    <string>R</string>
   </property>
  </action>
  <action name="action_Stop">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../../QModules.qrc">
     <normaloff>:/QModules/icons/Dialog-STOP.svg</normaloff>:/QModules/icons/Dialog-STOP.svg</iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="toolTip">
    <string>Stop recording</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="action_SilentMode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../QModules.qrc">
     <normaloff>:/QModules/icons/Audio-volume-muted.svg</normaloff>:/QModules/icons/Audio-volume-muted.svg</iconset>
   </property>
   <property name="text">
    <string>Silent mode</string>
   </property>
   <property name="toolTip">
    <string>Enable/disable spectrometer's silent mode</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header>QtCharts</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>Spectrum</tabstop>
  <tabstop>SBExposureTime</tabstop>
  <tabstop>SBLowerFrequency</tabstop>
  <tabstop>SBUpperFrequency</tabstop>
 </tabstops>
 <resources>
  <include location="../../QModules.qrc"/>
  <include location="../../DynExpManager.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>action_Save_CSV</sender>
   <signal>triggered()</signal>
   <receiver>SpectrumViewer</receiver>
   <slot>OnSaveCSVClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>295</x>
     <y>184</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>OnSaveCSVClicked()</slot>
 </slots>
</ui>
